segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/ZKSegmentContainerMonitor.java

```diff
@@ -196,51 +202,63 @@ private void checkAssignment() {
             return null;
         } else {
             this.pendingTasks.add(containerId);
-            return registry
-                    .stopContainer(handle, CLOSE_TIMEOUT_PER_CONTAINER)
-                    .whenComplete((aVoid, throwable) -> {
-                        if (throwable != null) {
-                            log.warn("Stopping container {} failed: {}", containerId, throwable);
-                        }
-                        try {
-                            // We remove the handle and don't attempt retry on stop container failures.
-                            unregisterHandle(containerId);
-                        } finally {
-                            // The pending task has to be removed after the handle is removed to avoid inconsistencies
-                            // with the container state.
-                            // Using finally block to ensure this is always called, otherwise this will prevent other
-                            // tasks from being attempted for this container id.
-                            this.pendingTasks.remove(containerId);
-                        }
-                    });
+            try {
+                return registry
+                        .stopContainer(handle, CLOSE_TIMEOUT_PER_CONTAINER)
```

Comment:
Currently we assume failed containers to be in stopped state. Need to review whether retrying stopContainer has any affect. Have filed #1537 to address this since its not in scope for this PR.